src/parameterizations/vertical/MOM_opacity.F90

```diff
@@ -1098,11 +1111,39 @@ subroutine opacity_init(Time, G, GV, US, param_file, diag, CS, optics)
                  default=PenSW_minthick_dflt, units="m", scale=GV%m_to_H)
   optics%PenSW_absorb_Invlen = 1.0 / (PenSW_absorb_minthick + GV%H_subroundoff)
 
+  ! The defaults for the following coefficients are taken from Manizza et al., GRL, 2005.
+  call get_param(param_file, mdl, "OPACITY_VALUES_MANIZZA", opacity_coefs, &
+                 "Pairs of opacity coefficients for blue, red and near-infrared radiation with "//&
+                 "parameterizations following the functional form from Manizza et al., GRL 2005, "//&
+                 "namely in the form opacity = coef_1 + coef_2 * chl**pow for each band.  Although "//&
+                 "coefficients are set for 3 bands, more or less bands may actually be used, with "//&
+                 "extra bands following the same properties as band 3.", &
+                 units="m-1", scale=US%Z_to_m, defaults=(/0.0232, 0.074, 0.225, 0.037, 2.86, 0.0/), &
+                 do_not_log=(CS%opacity_scheme/=MANIZZA_05))
+  call get_param(param_file, mdl, "CHOROPHYLL_POWER_MANIZZA", opacity_powers, &
+                 "Powers of chlorophyll for blue, red and near-infrared radiation bands in "//&
+                 "expressions for opacity of the form opacity = coef_1 + coef_2 * chl**pow.", &
+                 units="nondim", defaults=(/0.674, 0.629, 0.0/), &
+                 do_not_log=(CS%opacity_scheme/=MANIZZA_05))
+
+  ! The defaults for the following coefficients are taken from Morel and Antoine (1994).
+  call get_param(param_file, mdl, "OPACITY_VALUES_MOREL", extinction_coefs, &
+                 "Shortwave extinction length coefficients for shortwave radiation in the form "//&
+                 "proposed by Morel (1988), opacity = 1 / (sum(Coef(n) * log10(Chl)**(n-1))).", &
+                 units="m", scale=US%m_to_Z, defaults=(/7.925, -6.644, 3.662, -1.815, -0.218, 0.502/), &
+                 do_not_log=(CS%opacity_scheme/=MOREL_88))
+  call get_param(param_file, mdl, "SW_PEN_FRAC_COEFS_MOREL", sw_pen_frac_coefs, &
+                 "Coefficients for the shortwave radiation fraction in a fifth order polynomial "//&
+                 "fit as a function of log10(Chlorophyll).", &
+                 units="nondim", defaults=(/0.321,  0.008, 0.132,  0.038, -0.017, -0.007/), &
+                 do_not_log=(CS%opacity_scheme/=MOREL_88))
+
   if (.not.allocated(optics%min_wavelength_band)) &
     allocate(optics%min_wavelength_band(optics%nbands))
   if (.not.allocated(optics%max_wavelength_band)) &
     allocate(optics%max_wavelength_band(optics%nbands))
 
+  ! Set opacity scheme dependent parameters.
   if (CS%opacity_scheme == MANIZZA_05) then
     optics%min_wavelength_band(1) =0
```

Comment:
this could be close to 300nm unless we want to include gamma rays and UV

Reply:
Yes, it could be, but I think that is something we should deal with in a future PR.